### PR TITLE
[docs] init list of oss projects

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

List open source users.

image

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/index.rst [hunk 1]
```diff
@@ -195,6 +195,7 @@ Papers
 
    ray-overview/index.rst
    installation.rst
+   oss-users.rst
```
Comment (ericl) [inline doc/source/index.rst:199]: Should we put this under the community libraries section at the end? Like "More integrations" or something

File: doc/source/oss-users.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
+Open source projects using Ray
+==============================
+
+The Ray ecosystem is rapidly growing! This page lists projects and companies that are known to use Ray. This does not constitute an endorsement.
```
Comment (ericl) [inline doc/source/oss-users.rst:4]: Let's just include open source projects. It's not useful to users to know what companies are using ray, they can see ray.io for that.

File: doc/source/oss-users.rst [hunk 3]
```diff
@@ -0,0 +1,33 @@
+Open source projects using Ray
```
Comment (ericl) [inline doc/source/oss-users.rst:1]:
```suggestion
Open source libraries that run on Ray
```

File: doc/source/index.rst (hunk 1 above)
Comment (ericl) [reply doc/source/index.rst:199]: We can also link to this page from README, intro pages, etc.

Comment (richardliaw) [reply doc/source/index.rst:199]: hmm I think this is a big highlight though (think of like a webpage with endorsements)

Comment (ericl) [reply doc/source/index.rst:199]: Maybe we bring the entire libraries section up higher then

Comment (richardliaw) [reply doc/source/index.rst:199]: Screen Shot 2020-09-13 at 5 58 58 PM
currently looks like this. just worried that moving all libraries up will push down Ray Core things too much.

File: doc/source/oss-users.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
+.. _ray-oss-list:
+
+Open source projects that run on Ray
```
Comment (ericl) [inline doc/source/oss-users.rst:3]:
```suggestion
Open source libraries that integrate with Ray
```

File: doc/source/oss-users.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+====================================
+
+The Ray ecosystem is rapidly growing! This page lists projects that are known to use Ray. This does not constitute an endorsement.
```
Comment (ericl) [inline doc/source/oss-users.rst:6]:
```suggestion
The Ray ecosystem is rapidly growing! This page lists libraries that integrate with Ray for distributed execution. This does not constitute an endorsement.
```

File: doc/source/oss-users.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+*Using Ray?* `Fill out this form <https://forms.gle/ps4S1C955kjotxd57>`__ *and we'll be in contact to get added to the list.*
```
Comment (ericl) [inline doc/source/oss-users.rst:8]:
```suggestion
*Using Ray?* `Fill out this form <https://forms.gle/ps4S1C955kjotxd57>`__ *and we'll be in contact to get added to the list.*
```

File: doc/source/oss-users.rst [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+Mars is a tensor-based unified framework for large-scale data computation which scales Numpy, Pandas and Scikit-learn.
+
+Github: `https://github.com/mars-project/mars/ <https://github.com/mars-project/mars/>`_
+
+Spacy
+-----
+spaCy is a library for advanced Natural Language Processing in Python and Cython. It's built on the very latest research, and was designed from day one to be used in real products.
+
+Github: `https://github.com/explosion/spaCy/ <https://github.com/explosion/spaCy/>`_
```
Comment (ericl) [inline doc/source/oss-users.rst:28]: Can we link to their documentation on Ray integration, here and elsewhere? The generic project link isn't that useful, it's easy enough to find by googling.

File: doc/source/pandas_on_ray.rst [hunk 8]
```diff
@@ -1,5 +1,5 @@
-Pandas on Ray
-=============
+Modin (Pandas on Ray)
```
Comment (ericl) [inline doc/source/pandas_on_ray.rst:1]: Please keep this the same for consistency with other X on Ray libraries.

File: doc/source/ray-overview/basics.rst [hunk 9]
```diff
@@ -25,3 +25,5 @@ Ray also has a number of other community contributed libraries:
 - :doc:`../pandas_on_ray`
 - :doc:`../joblib`
 - :doc:`../multiprocessing`
+
+Check out the :ref:`rapidly growing list of open source projects <ray-oss-list>` that are known to use Ray!
```
Comment (ericl) [inline doc/source/ray-overview/basics.rst:29]:
```suggestion
Check out the :ref:`full list of distributed libraries <ray-oss-list>` that integrate with Ray.
```

Tone.

File: doc/source/ray-overview/basics.rst [hunk 10]
```diff
@@ -25,3 +25,5 @@ Ray also has a number of other community contributed libraries:
 - :doc:`../pandas_on_ray`
 - :doc:`../joblib`
 - :doc:`../multiprocessing`
```
Comment (ericl) [inline doc/source/ray-overview/basics.rst:27]: These aren't community contributed, we should remove joblib and multiprocessing.

File: doc/source/ray-overview/basics.rst (hunk 9 above)
Comment (ericl) [reply doc/source/ray-overview/basics.rst:29]: Actually I think we should replace this entire section (this sentence, and the bullets above), with:

```
There are also many `community libraries` that integrate with Ray, including `Modin`, `Dask`,
`MARS`, `Horovod`, `Flambe`, and others. Check out Ray's `full list of distributed libraries`.
```

Where both community and full list link to the community libraries page. We can pick perhaps the top 5 most popular libraries to link by name (not sure which makes sense here). Then, this snippet can be easily reused not only here but in the README, intro, etc.

File: doc/source/oss-users.rst (hunk 6 above)
Comment (richardliaw) [reply doc/source/oss-users.rst:8]: there doesn't seem to be a change here suggested so i'll just resolve

File: doc/source/oss-users.rst (hunk 7 above)
Comment (richardliaw) [reply doc/source/oss-users.rst:28]: I've added integration docs for those that make sense.

**Comment by richardliaw** [comment]
Note that the sidebar has a smaller amount of text than the page title. This is because it is too verbose otherwise.

File: doc/source/index.rst [hunk 11]
```diff
@@ -195,6 +195,7 @@ Papers
 
    ray-overview/index.rst
    installation.rst
+   Open source projects that run on Ray <oss-users.rst>
```
Comment (ericl) [inline doc/source/index.rst:198]:
```suggestion
   Open source libraries that run on Ray <oss-users.rst>
```

File: doc/source/index.rst [hunk 12]
```diff
@@ -195,6 +195,7 @@ Papers
 
    ray-overview/index.rst
    installation.rst
+   Community libraries that run on Ray <oss-users.rst>
```
Comment (ericl) [inline doc/source/index.rst:198]:
```suggestion
   oss-users.rst
```

File: doc/source/oss-users.rst [hunk 13]
```diff
@@ -0,0 +1,80 @@
+.. _ray-oss-list:
+
+Community libraries that integrate with Ray
+===========================================
+
+This page lists libraries that integrate with Ray for distributed execution. If you'd like to add your project to this list, feel free to file a pull request or open an issue on GitHub.
+
+*Using Ray?* `Fill out this form <https://forms.gle/ps4S1C955kjotxd57>`__ *and we'll be in contact to get added to the list.*
```
Comment (ericl) [inline doc/source/oss-users.rst:8]:
```suggestion
```

File: doc/source/ray-overview/basics.rst [hunk 14]
```diff
@@ -18,10 +18,10 @@ On top of **Ray Core** are several libraries for solving problems in machine lea
 - :ref:`sgd-index`
 - :ref:`rayserve`
 
-Ray also has a number of other community contributed libraries:
+There are also many :ref:`community libraries <ray-oss-list>` that integrate with Ray, including `Modin`_, `Hugging Face Transformers`_, :doc:`../dask-on-ray`, `MARS`_, `Horovod`_ and others.
```
Comment (ericl) [inline doc/source/ray-overview/basics.rst:21]:
```suggestion
There are also many :ref:`community libraries <ray-oss-list>` that integrate with Ray, including `Modin`_, `Hugging Face Transformers`_, :doc:`../dask-on-ray`, `MARS`_, `Horovod`_, and others.
```

File: doc/source/ray-overview/basics.rst [hunk 15]
```diff
@@ -18,10 +18,10 @@ On top of **Ray Core** are several libraries for solving problems in machine lea
 - :ref:`sgd-index`
 - :ref:`rayserve`
 
-Ray also has a number of other community contributed libraries:
+There are also many :ref:`community libraries <ray-oss-list>` that integrate with Ray, including `Modin`_, `Hugging Face Transformers`_, :doc:`../dask-on-ray`, `MARS`_, `Horovod`_, and others.
+Check out Ray's :ref:`rapidly growing list of open source projects <ray-oss-list>`!
```
Comment (ericl) [inline doc/source/ray-overview/basics.rst:22]:
```suggestion
Check out the :ref:`full list of Ray distributed libraries <ray-oss-list>`!
```

File: doc/source/ray-overview/basics.rst [hunk 16]
```diff
@@ -18,10 +18,10 @@ On top of **Ray Core** are several libraries for solving problems in machine lea
 - :ref:`sgd-index`
 - :ref:`rayserve`
 
-Ray also has a number of other community contributed libraries:
+There are also many :ref:`community libraries <ray-oss-list>` that integrate with Ray, including `Modin`_, `Hugging Face Transformers`_, :doc:`../dask-on-ray`, `MARS`_, `Horovod`_, and others.
+Check out the :ref:`full list of Ray distributed libraries <ray-oss-list>`!
```
Comment (ericl) [inline doc/source/ray-overview/basics.rst:22]:
```suggestion
Check out the :ref:`full list of Ray distributed libraries here <ray-oss-list>`.
```

File: doc/source/ray-overview/basics.rst [hunk 17]
```diff
@@ -18,10 +18,10 @@ On top of **Ray Core** are several libraries for solving problems in machine lea
...
+Check out the :ref:`full list of Ray distributed libraries here <ray-oss-list>`.
 
-- :doc:`../dask-on-ray`
-- `Mars on Ray <https://github.com/mars-project/mars/pull/1508>`__
-- :doc:`../pandas_on_ray`
-- :doc:`../joblib`
-- :doc:`../multiprocessing`
+.. _`Modin`: https://github.com/modin-project/modin
+.. _`Hugging Face Transformers`: https://huggingface.co/transformers/main_classes/trainer.html#transformers.Trainer.hyperparameter_search
+.. _`MARS`: https://github.com/mars-project/mars/pull/1508
```
Comment (ericl) [inline doc/source/ray-overview/basics.rst:26]:
```suggestion
.. _`MARS`: mars-on-ray.html
```

**Review by ericl — APPROVED** [review]

**Pull request merged**